internal/smartyresponseprocessing/smartyresponseprocessing.go

```diff
@@ -0,0 +1,81 @@
+package smartyresponseprocessing
```

Comment:
IMO, it would be okay to fold this in to `smarty-usps-comparison` since it seems specific to that tool. If you're thinking about reusing this logic, though, makes sense as a separate package.

Reply:
I've opted to fold the `smartyresponseprocessing` package into `smarty-usps-comparison`. I agree that the logic is very tool-specific, so this change makes sense to me. However `main.go` seems a bit lengthy now. In addressing some of the other feedback, I've added a few utility functions which have contributed to its length (some more generic than others.) Would it make sense to create a util folder within `smarty-usps-comparison` to house these?

Reply:
If you thinks `main.go` is too long, one option is to make other `.go` files in the same package (directory). This is a good option if you don't need the modularity of separate packages (for reuse, information hiding, etc.) but there are logical divisions in the code. For example, the [fmt package](https://github.com/golang/go/tree/master/src/fmt) puts printing, scanning, and formatting each into its own file.

Another option is to split something out into a separate package (directory). In this case, that might make sense for the Smarty stuff. It's easy to imagine using the Smarty client outside this program. A side benefit is that the type names don't need `Smarty`, which is nicer IMO.

Reply:
Makes sense! If we decide to split some of the smarty-related functionality into a separate package, would belong within the internal directory, or would it be better as a standalone package? I see the potential for it's use outside of `smarty-usps-comparison`  but would it ever be useful outside of the usps module?

Reply:
If we wanted to create a Smarty package for other programs, we'd probably put it in its own module / repo. We'd also have to look harder at the Smarty SDK and decide whether it's worth investing in our own client. So for now, putting this in `internal` makes sense to me.